File: Fitch12-2012/Prob1.java
import java.util.*;

public class Prob1 {
    public static void main(String[] args) {
        Scanner in = new Scanner(System.in);
        for(int i = 0; i < 4; i++) {
            int a = in.nextInt();
            int b = in.nextInt();
            if(b < a) {
                int temp = b;
                b = a;
                a = temp;
            }
            if(Math.sqrt(Math.pow(a, 2) + Math.pow(b, 2)) == Math.round(Math.sqrt(Math.pow(a, 2) + Math.pow(b, 2)))) {
                System.out.println((int) Math.sqrt(Math.pow(a, 2) + Math.pow(b, 2)));
                continue;
            }
            if(Math.sqrt(Math.pow(b, 2) - Math.pow(a, 2)) == Math.round(Math.sqrt(Math.pow(b, 2) - Math.pow(a, 2)))) {
                if((int) Math.sqrt(Math.pow(b, 2) - Math.pow(a, 2)) != 0) {
                    System.out.println((int) Math.sqrt(Math.pow(b, 2) - Math.pow(a, 2)));
                    continue;
                }
            }
            System.out.println("NO");
        }
    }
}

File: Fitch12-2012/Prob1.py
import math

def main():
    for i in range(4):
        arr = list(map(int, input().strip().split(" ")))
        a = arr[0]
        b = arr[1]
        if b < a:
            temp = b
            b = a
            a = temp
        if math.sqrt(a * a + b * b) == math.floor(math.sqrt(a * a + b * b)):
            print(int(math.sqrt(a * a + b * b)))
        elif math.sqrt(b * b - a * a) == math.floor(math.sqrt(b * b - a * a)):
            print(int(math.sqrt(b * b - a * a)))
        else:
            print("NO")

if __name__ == "__main__": main()
